### PR TITLE
Fat 16665 add back

### PR DESCRIPTION
**Pull request by VladUgarov**
all the described functionality is implemented

### CODE REVIEW

File: client/frontend.js [hunk 1]
```diff
@@ -3,53 +3,96 @@ const request = new XMLHttpRequest();
...
+    const fileName = document.getElementById("readOnlyFileName").value
+    const json = JSON.stringify({fileName: fileName});
+    request.open('POST', '/api/readFile', true);
+    request.setRequestHeader('Content-Type', 'application/json');
+    request.addEventListener("load", () => {
+        const received = JSON.parse(request.response);
+        if (request.status === 200) {
+            document.getElementById('readOnlyFileContent').innerHTML = "";
+            document.getElementById('readOnlyFileContent').innerHTML = received.content;
+            clearServerResponse()
```
Comment (AlexTishchenkoAndersen) [inline client/frontend.js:43]: semicolon pls

File: client/frontend.js [hunk 2]
```diff
@@ -3,53 +3,96 @@ const request = new XMLHttpRequest();
...
+    const fileName = document.getElementById("readFileName").value;
+    const json = JSON.stringify({fileName: fileName});
+    request.open('POST', '/api/readFile', true);
+    request.setRequestHeader('Content-Type', 'application/json');
+    request.addEventListener("load", () => {
+        const received = JSON.parse(request.response);
+        if (request.status === 200) {
+            document.getElementById('updateFileContent').innerHTML = "";
+            document.getElementById('updateFileContent').innerHTML = received.content;
+            clearServerResponse()
```
Comment (AlexTishchenkoAndersen) [inline client/frontend.js:60]: semicolon

File: client/frontend.js [hunk 3]
```diff
@@ -3,53 +3,96 @@ const request = new XMLHttpRequest();
...
+function updateFile() {
+    const fileName = document.getElementById("readFileName").value;
+    const fileContent = document.getElementById("updateFileContent").value;
+    const json = JSON.stringify({fileName: fileName, fileContent: fileContent });
+    request.open('PUT', '/api/updateFile', true);
+    request.setRequestHeader('Content-Type', 'application/json');
+    request.addEventListener("load", () => {
+        const received = JSON.parse(request.response);
+        if (request.status === 200) {
+            clearServerResponse()
```
Comment (AlexTishchenkoAndersen) [inline client/frontend.js:76]: semicolon

File: client/frontend.js [hunk 4]
```diff
@@ -3,53 +3,96 @@ const request = new XMLHttpRequest();
...
+    });
+    request.send(json);
+};
 
+function getAllFiles() {
+    request.open('GET', '/api/getAllFiles', true);
+    request.addEventListener("load", () => {
+        const received = JSON.parse(request.response);
+        if (request.status === 200) {
+            alert(received)
```
Comment (AlexTishchenkoAndersen) [inline client/frontend.js:88]: ;

File: functions.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import path from 'path'
```
Comment (AlexTishchenkoAndersen) [inline functions.js:1]: ;

File: functions.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import fs from 'fs'
```
Comment (AlexTishchenkoAndersen) [inline functions.js:2]: ;

File: functions.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+export async function getAllFiles() {
+    return new Promise((resolve, reject) => {
+        fs.readdir(path.resolve('files'), function (error,files) {
+            if (error) {
+                reject(new Error("Getting the list of files ended with an error"));
+            } else {
+                const arrFiles =[]
+                files.forEach(file => {
+                    arrFiles.push(file)
```
Comment (AlexTishchenkoAndersen) [inline functions.js:72]: ;

File: functions.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    });
+}
+
+export async function getAllFiles() {
+    return new Promise((resolve, reject) => {
+        fs.readdir(path.resolve('files'), function (error,files) {
+            if (error) {
+                reject(new Error("Getting the list of files ended with an error"));
+            } else {
+                const arrFiles =[]
```
Comment (AlexTishchenkoAndersen) [inline functions.js:70]: ; & space

**Comment by VladUgarov** [comment]
notes fixed

**Pull request closed without merging**